src/main/java/delete_event_use_case/DeleteEventDataAccess.java

```diff
@@ -0,0 +1,33 @@
+package delete_event_use_case;
+
+import data_access.DataAccessDay;
+import data_access.DataAccessEvent;
+import data_access.WeekDataAccess;
+
+import java.util.HashMap;
+import java.util.ArrayList;
+
+/**
+ * Delete Event Data Access class.
+ * @author : Khizer Ahmad
+ * @created: November 19th 2022
+ * @since : 2022-11-21, Mon.
+ **/
+public class DeleteEventDataAccess extends WeekDataAccess implements DeleteEventDsGateway {
+
+    @Override
+    public boolean eventExists(int dayIndex, String eventTitle) {
+        DataAccessDay day = super.getDays().get(dayIndex);
+        return day.getEvents().containsKey(eventTitle);
+        //this should always return true if event is stored correctly
```

Comment:
Since the option to delete an event only exists if an event exists, isn't it unnecessary to check whether the event exists after requesting to delete it?